src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  isAlive() {
+    return this.health > 0;
+  }
 
+  checkDeath() {
+    if (!this.isAlive()) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
[CODE PERFOMANCE] - Try to avoid nesting loops (indexOf + splice), use filter instead. In this case, you are using `indexOf` to find the index of the animal in the `Animal.alive` array and then `splice` to remove it. This leads to a complexity of O(n) for `indexOf` and O(n) for `splice` which in total gives you O(2n) ~ O(n). Instead, you can use `filter` method which will also give you O(n) but it's more readable and less error-prone.